elastio-s3-changelog/cloudformation-single.yaml

```diff
@@ -0,0 +1,263 @@
+# This template enables the S3 changelog for a single bucket.
+AWSTemplateFormatVersion: "2010-09-09"
+Parameters:
+  BucketName:
+    Type: String
+    AllowedPattern: .{3,}
+    ConstraintDescription: Bucket name must be at least 3 characters long.
+    Description: S3 bucket name.
+
+  ScanExistingObjects:
+    Type: String
+    AllowedValues: ['true', 'false']
+    Default: 'false'
+    Description: >
+      If set to true, the full initial scan of the bucket will be performed.
+      If set to false, only new objects in the bucket will be scanned.
+
+  EnableQrts:
+    Type: String
+    AllowedValues: ['true', 'false']
+    Default: 'false'
+    Description: >
+      If set to true, enables quasi-realtime scanning.
+      If set to false, disables quasi-realtime scanning.
+
+  KeyPrefixes:
+    # Unfortunately, it's impossible to pass CommaDelimitedList to a nested stack,
+    # so we have to use String here, and then split it manually.
+    Type: String
+    AllowedPattern: ^(([^/,\s]+/)+)(,([^/,\s]+/)+)*$|^$
+    ConstraintDescription: Prefixes must end (but not start) with '/', and cannot contain spaces.
+    Description: >
+      Comma-delimited list of prefixes of objects to scan. If empty, all objects will be scanned.
+      Example: xyz/,foo/bar/
+
+Conditions:
+  ScanExistingObjects: !Equals [!Ref ScanExistingObjects, 'true']
+  EnableQrts: !Equals [!Ref EnableQrts, 'true']
+  NoKeyPrefixes: !Equals [!Ref KeyPrefixes, '']
+
+Resources:
+  queue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days
+      VisibilityTimeout: 43200 # 12 hours
```

Comment:
I'm yet not sure we need such a big visibility time out. According to @anelson scanning a single object should be rather fast because scan doesn't depend on the object size, right? Then maybe smth like 1 hour or 30 minutes would be a better value? 

We need @vsazhenyuk-softheme and @anelson opinion here.